src/archive.rs

```diff
@@ -203,12 +203,28 @@ impl<'a> Archive<'a> {
     }
 }
 
+bitflags! {
+    #[derive(Default)]
+    pub struct ArchiveFlags: u32 {
+        const VOLUME = native::ROADF_VOLUME;
+        const COMMENT = native::ROADF_COMMENT;
+        const LOCK = native::ROADF_LOCK;
+        const SOLID = native::ROADF_SOLID;
+        const NEW_NUMBERING = native::ROADF_NEWNUMBERING;
+        const SIGNED = native::ROADF_SIGNED;
+        const RECOVERY = native::ROADF_RECOVERY;
+        const ENC_HEADERS = native::ROADF_ENCHEADERS;
+        const FIRST_VOLUME = native::ROADF_FIRSTVOLUME;
+    }
+}
+
 #[derive(Debug)]
 pub struct OpenArchive {
     handle: NonNull<native::HANDLE>,
     operation: Operation,
     destination: Option<WideCString>,
     damaged: bool,
+    flags: ArchiveFlags,
```

Comment:
Maybe make this pub?

Reply:
I'd rather not expose struct internals though.